File: solutions/Question 242/Solution2.py
# Problem: 242. Valid Anagram
# Difficulty: Easy
# Topics: Hash Table, String, Sorting
# URL: https://leetcode.com/problems/valid-anagram/
# Runtime: 16 ms
# Memory: 18.5 MB
# Solved via CodePath Auto-Committer

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        a = sorted(s)
        b = sorted(t)
        return a == b
          

File: solutions/Question 217/Solution2.cpp
/*
 * Problem: 217. Contains Duplicate
 * Difficulty: Easy
 * Topics: Array, Hash Table, Sorting
 * URL: https://leetcode.com/problems/contains-duplicate/
 * Runtime: 73 ms
 * Memory: 111.4 MB
 * Solved via CodePath Auto-Committer
 */

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> seen;
        for(int x : nums){
            if(seen.count(x)) return true;
            seen.insert(x);
        }
        return false;
    }
};